datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -710,6 +710,12 @@ private void handleAssignmentChange(boolean isDatastreamUpdate) throws TimeoutEx
       retryHandleAssignmentChange(isDatastreamUpdate);
     }
 
+    if (_config.getEnableAssignmentTokens()) {
+      List<DatastreamGroup> stoppingDatastreamGroups =
+          fetchDatastreamGroupsWithStatus(Collections.singletonList(DatastreamStatus.STOPPING));
```

Comment:
I don't think this API will work for the follower. The followers do not listen to the datastream change list. So, it does not maintain the latest snapshot of all the datastreams.  

The other challenge is: if you will try to fetch it, it is going to be an extremely expensive operation and will put pressure on zk server.

Reply:
I'm not sure how else to distinguish stopping streams other than having the followers fetch the datastream status from ZK. Let me see if there are any workarounds.

Reply:
Consider setting listener and notification for followers.

Reply:
That will still be an expensive operation. What I did instead is I wrote code that figures out the stopping streams from the diff of the old vs new assignment. The idea is that if all the tasks belonging to a stream are removed, then most likely the stream is stopped. Follower coordinator will try to claim the token for such streams.